File: src/Python/501-600/559.MaximumDepthOfN-aryTree.py

# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Node
        :rtype: int
        """
        return 0 if root is None else (1 if len(root.children) == 0 else max(map(lambda x : self.maxDepth(x), root.children)) + 1) 
        
if __name__ == '__main__':
    solution = Solution()
    root = Node(1, [])
    print(solution.maxDepth(root))
    root = Node(1, [Node(1, [(Node(1, []))])])
    print(solution.maxDepth(root))
    root = Node(1, [])
    print(solution.maxDepth(root))
else:
    pass

File: src/Java/501-600/559.MaximumDepthOfN-aryTree.java



import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
};

class Solution {
    public int maxDepth(Node root) {
        if (root == null) {
            return 0;
        }
        LinkedList<Node> childlist = new LinkedList<>();
        childlist.addAll(root.children);
        int count = 1;
        while (childlist.size() != 0) {
            count += 1;
            LinkedList<Node> newlist = new LinkedList<>();
            for (Node child : childlist) {
                newlist.addAll(child.children);
            }
            childlist = newlist;
        }
        return count;
    }
}
